### ISSUE TITLE
Does not correctly identify variable creation

### ISSUE THREAD

**Issue by DazWilkin** [bash-lsp/bash-language-server#318]
Thank you for creating bash-lsp. I use LSPs with Golang, Rust etc. and have missed the functionality when writing Bash.

I was delighted when I inadvertently hovered over a variable in a script and its declaration was documented but these variable (re)declarations appear to not be identified correctly.

```bash
#!/usr/bin/env bash

X="Freddie"

X="Frederik"

function test () {
  local X="Dog"
  printf "${X}"
}
```

In the above, the use of `X` in `printf "${X}"` (IMO) refers to its definition in `local X="Dog"` but bash-lsp reports line #3 (`X="Freddie"`).

If I comment `local X="Dog"`, I expect (perhaps incorrectly) that the redefinition `X="Frederik"` should be used but it continues to refer back to line #3 (`X="Freddie"`)

**Comment by skovhus** [comment]
Thanks for reporting this and sorry for the late reply. This is clearly a bug. Let me know if you are interested in fixing it.

**Comment by DazWilkin** [comment]
@skovhus thank you for taking the time to reply.

Unfortunately, I don't have the relevant skills to implement the fix.

**Comment by jfcherng** [comment]
Both 3.2.0 and 3.2.1 don't reply to the `initialize` payload.

**Comment by skovhus** [comment]
A much better heuristic for symbol resolution has been released as `bash-language-server@4.2.0` and for vscode in the latest extension 1.24.0. It opens up for implementing renaming of symbols, and I also believe this solves the issue here. 

Please let me know if it works for you or if you have any feedback to the new version.
